inst/examples/06-publishing.Rmd

```diff
@@ -19,42 +19,103 @@ bookdown::publish_book(render = 'local')
+
+1. Start with a **bookdown** project linked to a remote GitHub repository that you can push/pull to from your local copy. 
+    
+:::{.rmdimportant latex=true}
+
+If you don't have an existing book, you can create a simple **bookdown** HTML book to practice with instead:
 
-# create a branch named gh-pages and clean up everything
-git checkout --orphan gh-pages
-git rm -rf .
++ Make a new repository on GitHub (<https://happygitwithr.com/new-github-first.html#make-a-repo-on-github-2>),
++ Make a new RStudio Project via git clone: (<https://happygitwithr.com/new-github-first.html#new-rstudio-project-via-git-clone>),
++ From your RStudio Project, run this command in your R console: `bookdown::create_gitbook()` or `bookdown::create_bs4_book()` (depending on the HTML output format you want),
++ Change anything you want (or change nothing) and commit (<https://happygitwithr.com/new-github-first.html#make-local-changes-save-commit-1>),
++ Push your local changes to GitHub (<https://happygitwithr.com/new-github-first.html#push-your-local-changes-to-github>).
 
-# create a hidden file .nojekyll
-touch .nojekyll
-git add .nojekyll
+:::
 
-git commit -m"Initial commit"
-git push origin gh-pages
+
+### Change the output directory of your book
+
+In your local **bookdown** project, open the `_bookdown.yml` file and change the output directory of your book to be `/docs` by adding this line:
+
+```yaml
+output_dir: "docs"
 ```
 
+Save this file and close it.
+
+### Tell GitHub Pages to bypass Jekyll
+
+Create a hidden, empty file named `.nojekyll` from within R in your **bookdown** project root directory:
+
+```
+file.create(".nojekyll")
+```
+
+If you are using RStudio, you may need to refresh the viewer pane to see this new file. You can also use the terminal to create this file with the command `touch .nojekyll`.
+
+This file is [necessary](https://github.blog/2009-12-29-bypassing-jekyll-on-github-pages/) because GitHub Pages supports Jekyll ([jekyllrb.com](http://jekyllrb.com)), a static website builder, by default. You need the `.nojekyll` file because bookdown sites uses files or directories that start with underscores, and these files have a special status within the Jekyll framework. Because the **bookdown** HTML output is a static, standalone website, you need to tell GitHub that your website should *not* be built via Jekyll. 
+
+### Build your book, push to GitHub
+
+Use whichever method from Chapter \@ref(build-the-book) you prefer, then push to send your local changes online to GitHub.
+
+:::{.rmdwarning latex=true}
+
+If you have previously added directories like `_book` and `_bookdown_files` to your remote GitHub repository, you'll want to take a few more steps to remove them:
+
+* Delete the folders locally
+
+* From the terminal: 
+  * `git rm -r --cache _book/`
+  * `git rm -r --cache _bookdown_files/`
+  * `git commit -m 'Remove the rendered book files'`
+  * `git push origin main`
+
+:::
```

Comment:
I think in this case you also want to delete them locally, at least `_book`, isn't it ? 

Your new book will be in `docs/`. 

But I guess it is easier for a user to do that itself than knowing how to remove from git.

Reply:
I was told by several on the tidyverse team that you had to do both (which is what this says)- so you mean only do step 1?

Reply:
No no what I am saying is that  `git rm -r --cache _book/` will not delete the `_book` folder. 
It will still be there but untracked. If you don't need it anymore (because you are using `docs/` now, it should be removed I think (`bookdown::clean_book(TRUE)`)

For `_bookdown_files`, I don't remember what it is used for. Doing `git rm -r --cache _bookdown_files/` will untrack eveything from Git, but the files will still be there on the system. 

I wasn't sure here if we are supposed to keep those or not. 

Anyway, if those command where proven useful in the past, then it is the right one to switch. Usually I would set up at the start of a project, so I don't really try to undo stuff. But just start fresh. 

Meaning: 
* `bookdown::clean_book(TRUE)` => start fresh bookdown project
* change config `output_dir` and other stuff
* `bookdown::render_book()` 
* Commit everything (new stuff and also result of first step deletion.)

The only time I am using `git rm -r --cache` if I am right is when I want to untrack something without deleting it in my worktree because I want to now ignore it in the future